config/site/config.js

```diff
@@ -1,3 +1,12 @@
 window.TLINK_CONFIG = {
-    "regionalRoadsApiUrl": "https://devapi.regionalroads.com/simplewfs/"
+  "regionalRoadsApiUrl": "https://devapi.regionalroads.com/simplewfs/",
```

Comment:
I don't think this is the appropriate place to add the basemap configuration. The purpose of this file is to contain environment (dev/test/prod) specific configuration options and actually gets stored as a ConfigMap in OpenShift (https://console.apps.silver.devops.gov.bc.ca/k8s/ns/b16795-dev/configmaps/truck-route-planner2-dev-site-config/). 

Updates to basemap config would require updating config maps in OpenShift which is probably more trouble than it's worth.

Reply:
Thanks. I don't see a better home for this configuration I want to add, so I think I need to create a home for it. I think the "/config" directory is a good location. I see there's already /config/key/route-planner.json, so I could make something similar, like /config/key/basemaps.js. I'm also not sure what "key" is about, so maybe /config/basemaps.js. 

Or, we could make it more generic and easier to extend to include other configuration. Maybe "/config/config.json", though that could be confusing given there's also /config/site/config.js. So far I like this idea best. What are your thoughts?

Reply:
Could we read config from the baseMaps tool in map.json?

Reply:
It looks like config data is scoped to the section in which it's defined. This config is consumed in "viewer-leaflet". 

I added config to the "viewer" section in map.json, but it doesn't seem to get integrated when I try to reference it using SMK.TYPE.Viewer.

I found what looks like a better home for the esri basemap IDs: I can add them to the prototype.basemap object defined in "viewer" as a new "id" attribute for each of the basemaps. I'll see if this works.

I'm still not sure where to put the esriApiToken. I may create a new .js file for it using a global like "window.TLINK_BASEMAP" (instead of json since I'm not sure how I'd integrate it for consumption by "viewer-leaflet").

Reply:
I found that adding an "id" attribute to elements of the "basemap" object collides with the natural id of each element, so instead I'm using "esriStyleName", with "style name" taken from ESRI API: https://esri.github.io/esri-leaflet/api-reference/layers/vector-basemap.html. In a coming commit, these basemap names will be moved out of config.js.